dffml/operation/dataflow.py

```diff
@@ -33,8 +54,10 @@ class RunDataFlowConfig:
             )
             for val_def in val_defs
         ]
-
     async with self.octx.parent(self.config.dataflow) as octx:
+        # Register the subflow with parent,so that parent flow can
+        # forward inputs of specifed defintions(in DataFlow.forward)
+        await self.octx.register_subflow(self.parent.op.instance_name, octx)
```

Comment:
Let's either now, or in the future, make this into one call, so instead of:

```python

    async with self.octx.parent(self.config.dataflow) as octx:
        # Register the subflow with parent,so that parent flow can
        # forward inputs of specifed defintions(in DataFlow.forward)
        await self.octx.register_subflow(self.parent.op.instance_name, octx)
```

It would be:

```python
    async with self.subflow(self.config.dataflow) as octx:
```

So `subflow` would need to be a method of `OperationImplementationContext`

Reply:
Yes!

Reply:
wait ,i'm confused,`OperationImplementationContext` or `OperationImplementationNetworkContext` ?

Reply:
In `OperationImplementationContext`

Reply:
You'll need to use [contextlib.asynccontextmanager](https://docs.python.org/3/library/contextlib.html?highlight=contextlib#contextlib.asynccontextmanager) because you need to `yield` within an `async def` function